docs/agentless.md

```diff
@@ -0,0 +1,64 @@
+# PingFederate Authentication API Agentless Integration
+
+### Overview
+You can modify the JavaScript Widget to allow the Agentless Integration Kit to work with the PingFederate Authentication API.
+
+### System requirements and dependencies
+* Agentless Integration Kit 2.0 or later
+* An authentication service that is configured to drop off authenticated user attributes to PingFederate
+
+### Setup
+
+1. Modify the `reference_id_required.hbs` template to include the credential fields that your authentication service requires. The widget renders this handlebar template when it encounters the `REFERENCE_ID_REQUIRED` state.
+
+1. Add an event handler that sends the user credentials to the authentication service. The event handler is triggered when the user submits the form on the template.
+
+	Example constructor additions:
+	```js
+  	this.registerAgentlessHandler = this.registerAgentlessHandler.bind(this);
+  	this.handleAgentlessSignOn = this.handleAgentlessSignOn.bind(this);
+   
+  	this.addEventHandler('REFERENCE_ID_REQUIRED', this.registerAgentlessHandler);
+	```
+	Example functions:
+	```js
+  	registerAgentlessHandler() {
+       	Array.from(document.querySelectorAll('[data-agentlessActionid]')).
+       	forEach(element => element.addEventListener('click', this.handleAgentlessSignOn));
+  	}
+ 	 
+  	handleAgentlessSignOn(evt) {}
+	```
+    
+	The authentication service should validate the credentials, then drop off user attributes with PingFederate. PingFederate will provide it with a new reference ID (`REF`) that is associated with the attributes.
+    
+ 1. Add the following to send the new reference ID (`REF`) to PingFederate.
+
+	```js
+  	this.store.dispatch('POST_FLOW', 'checkReferenceId', "{json payload with referenceId}")
+	```
+
+	This completes the action and allows PingFederate to continue to the next part of the authentication policy. 
+
+##### Getting the original reference ID
+If the authentication service needs to retrieve the sign-on context from PingFederate, it will require the original reference ID (`REF`) associated with the current state. You can get this reference ID by calling `this.store.getStore().pickupReferenceId`.
+
+
+##### Dropping off attributes
+The drop-off endpoint (`/ext/ref/dropoff`) does not support CORS. The authentication service must drop off the user attributes with a back-channel call.
+
+
+### Documentation
+
+For the latest documentation, see [Agentless Integration Kit](https://docs.pingidentity.com/bundle/integrations/page/ygj1563994984859.html) in the Ping Identity [Support Home](https://support.pingidentity.com/s/)
```

Comment:
Can we link to https://support.pingidentity.com and let it redirect to `/s/`? Might be a little more future proof.

Reply:
apparently this is the most stable URL for integrations docs, there is no format like the one you mentioned used by PF docs.

Reply:
Yeah looks like we just link to the integrations home in the Agentless readme, so I think that's fine.

I'm thinking we might want to remove the path on the Support Home link in case they change where the support link redirects to.

Reply:
sure, I can remove that if you think it'll be changing soon.

Reply:
Yeah I think it might be safer to just have the URL with no path.